boards/arm/lpcxpresso55s36/pinmux.c

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/drivers/pinctrl.h>
+#include "fsl_common.h"
+
+static int lpcxpresso55s36_pinmux_init(void)
+{
+	int err;
+
+	/* Declare pin configuration state for USB pin here */
+	PINCTRL_DT_DEFINE(DT_NODELABEL(usbfs));
```

Comment:
Is there any reason we could not declare the pin configuration state within the USB DC driver? Even though the driver does not use the device model, it should still be possible to locate this code there and call `pinctrl_apply_state` during init. If no pin control states are specified, the function will of course return an error, but other drivers like the [memc_mcux_flexspi](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/memc/memc_mcux_flexspi.c#L187) driver simply ignore the return code if it indicates the error was that no pin control state was present.

Reply:
Thank you, I will add to the USB driver.